Base/QTCore/qSlicerCoreApplication.cxx

```diff
@@ -452,6 +452,22 @@ void qSlicerCoreApplicationPrivate::init()
 
   this->createDirectory(q->extensionsInstallPath(), "extensions"); // Make sure the path exists
 
+  this->ApplicationLocaleName = "en_US";
+  this->ApplicationLocale = QLocale(ApplicationLocale);
+#ifdef Slicer_BUILD_I18N_SUPPORT
+  if (q->userSettings()->value("Internationalization/Enabled").toBool())
+    {
+    QString localeName = q->userSettings()->value("language", this->ApplicationLocaleName).toString();
```

Comment:
Yes, I think all those options could be useful, but I'm not sure which are strictly required.  It might be enough to have a command line option, but maybe also we could automatically inherit the OS locale, at least as a default.  I agree it's a new feature that needs discussion.

Reply:
New feature request filed: https://github.com/Slicer/Slicer/issues/6833